### ISSUE TITLE
Debug option "Visible Navigation" does not show GridMap navmeshes.

### ISSUE THREAD

**Issue by EzraT** [godotengine/godot#32244]
**Godot version:** 3.1.1 and recent master build.

**OS/device including version:** Linux Mint 19.2 Cinnamon 4.2.4

**Issue description:**
![Untitled](https://user-images.githubusercontent.com/781121/65380330-92c66400-dcd9-11e9-9957-c6ff3c406566.jpg)
As the title states, this debug option does not work for Gridmap Navmeshes.
Works fine for Navmeshes outside of Gridmaps.

**Steps to reproduce:**
1: Clone this demonstration project: https://github.com/DigitalKI/GodotGridMapNav and load it up.
2: Enable the option as shown above and test.

**Comment by codecustard** [comment]
It seems to work when you bake the NavMesh.

**Comment by EzraT** [comment]
Hi @codecustard,

Could you elaborate on how exactly you baked your Gridmap?
I tried baking my gridmap with make_baked_meshes(), but it still does not show anything for me.

**Comment by codecustard** [comment]
Select the navmesh that your grid map is in and there should be a button at the top that says bake

**Comment by EzraT** [comment]
@codecustard Oh, I'm talking about the Navmeshes that you can pack into the MeshLibrary itself, not seperate NavigationMeshInstances outside of the GridMap that you bake afterwards.

**Comment by jonbonazza** [comment]
Unless I am misunderstanding your issue, you can't really connect navmeshes from individual meshes to create a navmesh for the whole grid map. Thus might be possible in 4.0 using a postprocessing script.

**Comment by EzraT** [comment]
> Unless I am misunderstanding your issue, you can't really connect navmeshes from individual meshes to create a navmesh for the whole grid map. Thus might be possible in 4.0 using a postprocessing script.

That's incorrect, Godot has 2 options for navmeshes using Gridmaps, you can decide to just bake a Navmesh from the Gridmaps geometry or colliders in the editor, but it also allows you to manually add navmeshes to the MeshLibrary itself, and then use those in the Gridmap.
I've tried that before and it works fine if the navmeshes inside the library connect to each other correctly. (vertex wise)

The problem I'm describing has to do with the debug option only showing NavmeshInstances outside of the gridmap itself, it does not show Navmeshes used inside a MeshLibrary.

**Comment by smix8** [comment]
GridMap has no function / check for debug_navigation hint like other navigation nodes so no debug meshes are created. Those debug meshes from outside for the GridMap are not part of the GridMap but created by the NavigationRegion navmesh bake. Noted this issue for https://github.com/godotengine/godot-proposals/issues/4507 rework.
